src/effection/events.ts

```diff
@@ -25,14 +25,21 @@ export function watch(
   prepare: (...args: any[]) => any = defaultPrepareMessage // eslint-disable-line @typescript-eslint/no-explicit-any
 ): Operation {
   return ({ resume, context: { parent }}) => {
+    let parentContext = parent as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
     for(let name of [].concat(names)) {
-      let context = parent as any; // eslint-disable-line @typescript-eslint/no-explicit-any
-      context.spawn(monitor(function*() {
-        while (true) {
-          let args = yield on(emitter, name);
-          yield send(prepare.apply({ event: name}, args), parent);
-        }
-      }));
+      let listener = (...args) => {
+        parentContext.spawn(fork(function*() {
```

Comment:
We don't have to wrap this operation in a generator here:

```ts
let deliverMessage = fork(send(prepare.apply({ event: name}, args), parentContext))
parentContext.spawn(deliverMessage);
```
Would also work.

Reply:
Neat :)